MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -831,20 +820,28 @@ namespace WPEFramework {
                         /* we set this to false */
                         g_is_critical_maintenance="false";
 
+
+                        if(m_thread.joinable()){
+                             m_thread.join();
+                        }
+
                         m_thread = std::thread(&MaintenanceManager::task_execution_thread, _instance);
 
                         result=true;
                     }
                     else {
                         LOGINFO("Already a maintenance is in Progress. Please wait for it to complete !!");
                     }
+                    m_statusMutex.unlock();
                     returnResponse(result);
                 }
 
         void MaintenanceManager::onMaintenanceStatusChange(Maint_notify_status_t status) {
             JsonObject params;
             /* we store the updated value as well */
-            g_notify_status=status;
+            m_statusMutex.lock();
+            m_notify_status=status;
```

Comment:
As commented before, you can remove this status update here as the class variable is updated and controlled by mutex.